src/kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -946,25 +956,24 @@ impl<'tcx> GotocCtx<'tcx> {
     /// This function computes the size and alignment of a dynamically-sized type.
     /// The implementations follows closely the SSA implementation found in
     /// rustc_codegen_ssa::glue::size_and_align_of_dst.
-    fn size_and_align_of_dst(&self, t: Ty<'tcx>, arg: Expr) -> SizeAlign {
+    ///
+    /// This returns an option because of issue with codegen dynamic T.
+    /// See https://github.com/model-checking/kani/issues/1074 for more details.
+    fn size_and_align_of_dst(&self, t: Ty<'tcx>, arg: Expr) -> Option<SizeAlign> {
         let layout = self.layout_of(t);
         let usizet = Type::size_t();
         if !layout.is_unsized() {
             let size = Expr::int_constant(layout.size.bytes_usize(), Type::size_t());
             let align = Expr::int_constant(layout.align.abi.bytes(), usizet);
-            return SizeAlign { size, align };
+            return Some(SizeAlign { size, align });
         }
         match t.kind() {
             ty::Dynamic(..) => {
-                //TODO figure out if this needs to be done, or if the result we want here is for the fat pointer.
-                //We need to get the actual value from the vtable like in codegen_ssa/glue.rs
-                // let vs = self.layout_of(self.tcx.vtable_methods(binder.principal().unwrap().with_self_ty(self.tcx, t)));
-                // https://rust-lang.github.io/unsafe-code-guidelines/layout/pointers.html
-                // The size of &dyn Trait is two words.
-                let size = Expr::int_constant((layout.size.bytes_usize()) * 2, Type::size_t());
-                // The alignment of &dyn Trait is the word size.
-                let align = Expr::int_constant(layout.align.abi.bytes(), usizet);
-                SizeAlign { size, align }
+                // TODO: We don't support size of dynamic yet.
+                // https://github.com/model-checking/kani/issues/1074
+                // The size of dyn T is only known at runtime. Thus, we need to retrieve that
+                // information from the object vtable.
+                None
```

Comment:
Does `arg.member("vtable").member("size")` not work here?

Reply:
Probably. I just didn't have time to go figure it out. Let me try and maybe we can just fix this issue. :)